### PR TITLE
fix: js/css are minified when both minify.js and minify.css are false

### PR DESCRIPTION
**Pull request by xc2**
## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: packages/core/src/helpers.ts
Comment (chenjiahan) [inline packages/core/src/helpers.ts:1]: Can we add some PR descriptions 😄

File: packages/core/src/helpers.ts [hunk 1]
```diff
@@ -617,3 +622,36 @@ export const prettyTime = (seconds: number): string => {
...
+
+  if (minify === true) {
+    return {
+      minifyJs: true,
+      minifyCss: true,
+    };
+  }
+
+  return {
+    minifyJs: !(minify.js === false || !isProd),
```
Comment (chenjiahan) [inline packages/core/src/helpers.ts:652]: I think `minify.js` can be enabled in dev mode

Comment (chenjiahan) [reply packages/core/src/helpers.ts:652]: My mistake, forget it..

File: packages/core/src/plugins/minimize.ts [hunk 2]
```diff
@@ -99,22 +63,17 @@ export const pluginMinimize = (): RsbuildPlugin => ({
...
-      const { SwcJsMinimizerRspackPlugin, LightningCssMinimizerRspackPlugin } =
-        rspack;
+      const { minifyJs, minifyCss, jsOptions, cssOptions } = parseMinifyOptions(
+        config,
+        isProd,
+      );
 
-      const { minifyJs, minifyCss, jsOptions, cssOptions } =
-        parseMinifyOptions(config);
+      chain.optimization.minimize(minifyJs || minifyCss);
```
Comment (chenjiahan) [inline packages/core/src/plugins/minimize.ts:71]: `chain.optimization.minimize(minifyJs || minifyCss);` is set by the basic plugin, should we set it here again?

Comment (xc2) [reply packages/core/src/plugins/minimize.ts:71]: > `chain.optimization.minimize(minifyJs || minifyCss);` is set by the basic plugin, should we set it here again?

it is for a such edge case that `pluginBasic` is not applied - typically seen in test cases.

actually i'm not sure why it's in `pluginBasic`

Comment (chenjiahan) [reply packages/core/src/plugins/minimize.ts:71]: Yeah we can also remove the `chain.optimization.minimize()` from basic plugin.

But this SWC/LightningCSS minimizer plugin is skipped in the webpack mode 😂, so we need to set ``chain.optimization.minimize()` before this `if` condition.

image

Comment (xc2) [reply packages/core/src/plugins/minimize.ts:99]: i've refined this PR, please check again.

File: packages/core/src/plugins/minimize.ts [hunk 3]
```diff
@@ -64,25 +64,20 @@ export const parseMinifyOptions = (
...
-    return {
-      minifyJs: false,
-      minifyCss: false,
-    };
+  if (!isProd) {
+    return { minifyJs: false, minifyCss: false };
   }
+  const { minify: _minify } = config.output;
+  const minify =
+    _minify && typeof _minify === 'object'
```
Comment (chenjiahan) [inline packages/core/src/plugins/minimize.ts:72]: I prefer to use the former implementation because it seems simpler. 😂

Comment (xc2) [reply packages/core/src/plugins/minimize.ts]: Done!

**Review by chenjiahan — APPROVED** [review]
